### PR TITLE
Show 'Played Time this Level'

### PR DESCRIPTION
**Pull request by tflo**
This little PR adds the **Time Played this Level** to the char tooltip lines.

The additional resource usage is practically zero, since timePlayedThisLevel comes "for free" as the second argument of the TIME_PLAYED_MSG event.

---

### Reasoning

This stat is not too interesting while you are leveling a char, but I find it becomes very interesting once you have all your chars (or all your favorite chars) at max level: It basically approximates the *time played in the current expansion*, as opposed to the more historical Total Time Played.

For example, I have a char that is absolutely leading the _Total Time Played_ charts, simply because it was the first one. However, when I look at *Time Played this Level*, I see that there are a couple of chars that have more days than the supposed leader. Having both stats at a glance lets me see my tendency in terms of class preference.

---

### Some Details

The stat is displayed along with the char level, and also follows the user setting for displaying char levels:

pt-main

---

You can also sort by the new stat:

pt-sort

---

I have manipulated your `FormatTime` function to return a "?" if the first parameter is nil. This seemed to be the most elegant way to handle the inevitable nil that comes from existing char subtables where the `timePlayedLevel` key does not yet exist:

pt-unknown

So, basically the "?" means 'Login to that char to get the data'. But this was a bit too long for my taste, and I guess the user will get the meaning sooner or later 😉 

---

### Notes

I discovered your addon only yesterday and have not familiarized myself much with the code, i.e. my implementation is the straightforward type. Also, I have only tested it for a few hours (but with a good number of chars). So – if you like it – please take it as a suggestion, I'm sure there is room for improvement.

I made your `t` variable local (line 792 in the PR), since I didn't see any reason why it should be global. If I missed something, please revert that.

---

Thanks for considering.

### CODE REVIEW

**Comment by LudiusMaximus** [comment]
Ups, this PR completely evaded me. Apparently I had notifications deactivated. Sorry.
Now, there are merge conflicts as I have just made many changes.
Would you mind resolving these and then renew the PR?

**Comment by tflo** [comment]
I don’t have write access here, but I can (try to) merge your CurseForge files with my local repo (where I have better merge tools anyway), and then update my PR. 

I assume the CurseForge version (11.0.7.0) is up to date?

**Comment by LudiusMaximus** [comment]
Yes, exactly! Thanks!

**Comment by tflo** [comment]
Updated PR here: https://github.com/LudiusMaximus/Broker_PlayedTime/pull/8

Please read the comment.
